.github/workflows/ci.yml

```diff
@@ -0,0 +1,80 @@
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+name: Continuous integration
+
+jobs:
+  fmt:
+    name: Rustfmt
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install tooling
+        run: |
+          sudo apt-get install -y protobuf-compiler
+          protoc --version
+      - name: Install Rustfmt
+        run: rustup component add rustfmt
+      - name: Run Rustfmt
+        run: cargo fmt --all -- --check
+
+  toml-sort:
+    name: TOML Sort check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: Off-Narrative-Labs/toml_sort@v1
+        with:
+          all: true
+
+  test:
+      name: Test and code coverage
+      needs: [fmt, toml-sort]
+      runs-on: ubuntu-latest
+      permissions:
+          issues: write
+          pull-requests: write
+      steps:
+        - uses: actions/checkout@v4
+        - name: Install tooling
+          run: |
+            sudo apt-get install -y protobuf-compiler
+            protoc --version
+        - name: Install latest nextest release
+          uses: taiki-e/install-action@nextest
+        - name: Install cargo-llvm-cov
+          uses: taiki-e/install-action@cargo-llvm-cov
+        - name: Rust Cache
+          uses: Swatinem/rust-cache@v2
+        - name: Run tests
+          run: cargo llvm-cov nextest --lcov
+               --ignore-filename-regex "node/"
+               --output-path lcov.info
+        - if: github.event_name == 'pull_request'
+          name: Report code coverage
+          uses: Nef10/lcov-reporter-action@v0.4.0
+          with:
+            lcov-file: lcov.info
+            pr-number: ${{ github.event.pull_request.number }}
+
+  clippy:
+    name: Clippy
+    needs: [fmt, toml-sort]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install tooling
+        run: |
+          sudo apt-get install -y protobuf-compiler
+          protoc --version
+      - name: Install Clippy
+        run: rustup component add clippy
```

Comment:
If we have the components listed in the `rust-toolchain.toml` file can we remove this?

Reply:
It's working without it, not sure if it's because the components but I'll leave them there 😅